projects/experimental/directives/sensitive/sensitive.directive.ts

```diff
@@ -15,9 +25,18 @@ export class TuiSensitiveDirective {
     tuiSensitive: boolean | null = false;
 
     readonly offset = Math.round(Math.random() * 10) * 10;
+    readonly height$ = this.resize$.pipe(
+        map(([entry]) => entry.contentRect.height),
+        map(height => {
+            const rows = Math.max(2, Math.floor(height / 16) + 1);
+
+            return height * (rowsInSvg / rows);
+        }),
+    );
 
     constructor(
         @Inject(TuiDirectiveStylesService) directiveStyles: TuiDirectiveStylesService,
+        @Inject(TuiResizeService) private readonly resize$: TuiResizeService,
```

Comment:
In 4.0 we can use resize observer from ng-web-apis directly, we will drop this service and there will be no need for destroy service. If it is intended to go into 3.0 too then let's keep it like this for now.

Reply:
Yes, ok I'll refactor this after cherry-picking in 3.0